From ea0eb08a17866f1bc586f0c3d5c2a44babd7a67f Mon Sep 17 00:00:00 2001
From: Petr Vorel <pvorel@suse.cz>
Date: Tue, 20 Sep 2022 21:30:12 +0200
Subject: [PATCH 1/2] ping: Add SA_RESTART to sa_flags

To avoid missing output in write on aarch64.
This error was in ping since ever (since git era),
but got visible only after 4655ecc ("libcommon: check standard streams
status at exit").

Issue found with LTP net stress interface tests.

Fixes: 3337034 ("Initial import of iputils")
Fixes: https://github.com/iputils/iputils/issues/424

Reported-by: Petr Vorel <pvorel@suse.cz>
Suggested-by: Benjamin Poirier <benjamin.poirier@gmail.com>
Suggested-by: Cyril Hrubis <chrubis@suse.cz>
Reviewed-by: Cyril Hrubis <chrubis@suse.cz>
Signed-off-by: Petr Vorel <pvorel@suse.cz>
[ upstream status: https://github.com/iputils/iputils/commit/09f1b3706e90e748910c82612bc9d28fbfe0e9fb ]
---
 ping/ping.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ping/ping.h b/ping/ping.h
index 007ae28..3fb5156 100644
--- a/ping/ping.h
+++ b/ping/ping.h
@@ -316,6 +316,7 @@ static inline void set_signal(int signo, void (*handler)(int))
 	memset(&sa, 0, sizeof(sa));
 
 	sa.sa_handler = (void (*)(int))handler;
+	sa.sa_flags = SA_RESTART;
 	sigaction(signo, &sa, NULL);
 }
 
-- 
2.37.3

